nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/registry/flow/ImportAllFlows.java

```diff
@@ -0,0 +1,298 @@
+        return getVersionedFlows(client, bucketMap)
+                .stream()
+                .collect(Collectors.toMap(e -> new ImmutablePair<>(e.getBucketIdentifier(), e.getName()),
+                        VersionedFlow::getIdentifier));
+    }
+
+    private List<VersionedFlow> getVersionedFlows(final NiFiRegistryClient client, final Map<String, String> bucketMap) throws ParseException, IOException, NiFiRegistryException {
+        final List<VersionedFlow> flows = new ArrayList<>();
+        for (final String id : bucketMap.values()) {
+            final Properties flowProperties = new Properties();
+            flowProperties.setProperty(CommandOption.BUCKET_ID.getLongName(), id);
+
+            flows.addAll(listFlows.doExecute(client, flowProperties).getResult());
+        }
+        return flows;
+    }
+
+    private Map<String, List<Integer>> getVersionMap(final NiFiRegistryClient client, final Map<Pair<String, String>, String> flowMap,
+                                                     final  boolean isInteractive) throws ParseException, IOException, NiFiRegistryException {
+        printMessage(isInteractive, ALL_FLOW_VERSIONS_COLLECTED);
+        return getVersionedFlowSnapshotMetadataList(client, flowMap)
+                .stream()
+                .collect(Collectors.groupingBy(VersionedFlowSnapshotMetadata::getFlowIdentifier,
+                        Collectors.mapping(VersionedFlowSnapshotMetadata::getVersion, Collectors.toList())));
+    }
+
+    private List<VersionedFlowSnapshotMetadata> getVersionedFlowSnapshotMetadataList(final NiFiRegistryClient client,
+                                                                                     final Map<Pair<String, String>, String> flowMap) throws ParseException, IOException, NiFiRegistryException {
+        final List<VersionedFlowSnapshotMetadata> versions = new ArrayList<>();
+        for (final String flowIds : flowMap.values()) {
+            final Properties flowVersionProperties = new Properties();
+            flowVersionProperties.setProperty(CommandOption.FLOW_ID.getLongName(), flowIds);
+
+            versions.addAll(listFlowVersions.doExecute(client, flowVersionProperties).getResult());
+        }
+        return versions;
+    }
+
+    private List<String> getFilePathList(final Properties properties) throws MissingOptionException, NiFiRegistryException {
+        final String directory = getRequiredArg(properties, CommandOption.INPUT_SOURCE);
+        final List<String> files;
+
+        try (final Stream<Path> paths = Files.list(Paths.get(directory))) {
+            files = paths
+                    .filter(Files::isRegularFile)
+                    .filter(path -> path.getFileName().toString().startsWith(FILE_NAME_PREFIX))
+                    .map(Path::toString)
+                    .collect(Collectors.toList());
+        } catch (Exception e) {
+            throw new NiFiRegistryException("File listing not possible due to ", e);
```

Comment:
This approach to error messages needs to be changed. The message should stand by itself, it will not be concatenated with the exception. Recommend the following:
```suggestion
            throw new NiFiRegistryException("File listing failed", e);
```